src/kademlia/dht_state.cpp

```diff
@@ -78,29 +106,28 @@ namespace {
 
 		if (e.type() != bdecode_node::dict_t) return ret;
 
-		ret.nid = extract_node_id(e, "node-id");
-#if TORRENT_USE_IPV6
-		ret.nid6 = extract_node_id(e, "node-id6");
-#endif
+		ret.nids = extract_node_ids(e, "node-id");
 
 		if (bdecode_node const nodes = e.dict_find_list("nodes"))
 			ret.nodes = detail::read_endpoint_list<udp::endpoint>(nodes);
-#if TORRENT_USE_IPV6
-		if (bdecode_node const nodes = e.dict_find_list("nodes6"))
-			ret.nodes6 = detail::read_endpoint_list<udp::endpoint>(nodes);
```

Comment:
Good catch. I waffled a bit on whether to consolidate these into a single nodes list or not. ultimately I left them separate but missed reverting this change.